webui/webui/components/navbar.py

```diff
@@ -1,68 +1,124 @@
+                on_click=State.delete_chat,
+                stroke_width=1,
+            ),
+            width="20%",
+            variant="surface",
+            color_scheme="red",
+        ),
+        width="100%",
+    )
+
+
+def sidebar(trigger) -> rx.Component:
+    """The sidebar component."""
+    return rx.drawer.root(
+        rx.drawer.trigger(trigger),
+        rx.drawer.overlay(),
+        rx.drawer.portal(
+            rx.drawer.content(
+                rx.vstack(
+                    rx.heading("Chats", color=rx.color("mauve", 11)),
+                    rx.divider(),
+                    rx.foreach(State.chat_titles, lambda chat: sidebar_chat(chat)),
+                    align_items="stretch",
+                    width="100%",
                 ),
-                rx.chakra.link(
-                    rx.chakra.box(
-                        rx.chakra.image(src="favicon.ico", width=30, height="auto"),
-                        p="1",
-                        border_radius="6",
-                        bg="#F0F0F0",
-                        mr="2",
-                    ),
-                    href="/",
+                top="auto",
+                right="auto",
+                height="100%",
+                width="20em",
+                padding="2em",
+                background_color=rx.color("mauve", 2),
+                outline="none",
+            )
+        ),
+        direction="left",
+    )
+
+
+def modal(trigger) -> rx.Component:
+    """A modal to create a new chat."""
+    return rx.dialog.root(
+        rx.dialog.trigger(trigger),
+        rx.dialog.content(
+            rx.hstack(
+                rx.input(
+                    placeholder="Type something...",
+                    on_blur=State.set_new_chat_name,
+                    width="30em",
                 ),
-                rx.chakra.breadcrumb(
-                    rx.chakra.breadcrumb_item(
-                        rx.chakra.heading("ReflexGPT", size="sm"),
-                    ),
-                    rx.chakra.breadcrumb_item(
-                        rx.chakra.text(State.current_chat, size="sm", font_weight="normal"),
+                rx.dialog.close(
+                    rx.button(
+                        "Create chat",
+                        on_click=State.create_chat,
                     ),
                 ),
+                background_color=rx.color("mauve", 1),
+                spacing="2",
+                width="100%",
             ),
-            rx.chakra.hstack(
-                rx.chakra.button(
-                    "+ New chat",
-                    bg=styles.accent_color,
-                    px="4",
-                    py="2",
-                    h="auto",
-                    on_click=State.toggle_modal,
+        ),
+    )
+
+
+def navbar():
+    return rx.box(
+        rx.hstack(
+            rx.hstack(
+                rx.avatar(fallback="CC", variant="solid"),
+                rx.heading("Coca-Cola"),
```

Comment:
Do we want to merge this in as is? Maybe have Reflex by default and make it easy to customize the theme